### PR TITLE
Used Flask-RESTful for API

### LINKED ISSUE #162: REST API
Could you add REST API?

http://flask-restful.readthedocs.org

### PR DESCRIPTION
**Pull request by gmist**
Closes #162

### CODE REVIEW

File: main/api/fields.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class IntegerField(fields.Raw):
+  def format(self, value):
+    if value > 9007199254740992 or value < -9007199254740992:
+      return repr(value)
+    return value
+
+
+class KeyField(fields.Raw):
+  def format(self, value):
+    return value.urlsafe()
```
Comment (kaylarose) [inline main/api/fields.py]: Would that work for ```StructuredProperty``` and ```LocalStructuredProperty``` as well?

Comment (gmist) [inline main/api/fields.py]: Good question. For such cases Flask-RESTful has ```fields.Nested```. For example:

``` python
class A(ndb.Model):
  b = ndb.FloatProperty(default=12.12)
  bb = ndb.IntegerProperty(default=12)
  bbb = ndb.StringProperty(default='bbb')

class Config(model.Base, model.ConfigAuth):
  ...
  test_a = ndb.StructuredProperty(A)
  def _pre_put_hook(self):
    self.test_a = A()

A_FIELDS = {
  'b': fields.Float,
  'bb': fields.Integer,
  'bbb': fields.String,
}


CONFIG_FIELDS = {
  ...
  'test_a': fields.Nested(A_FIELDS)
}
```
...
screen shot 2015-01-01 at 21 56 59

**Comment by lipis** [comment]
What do you think about this sort of refactoring: https://github.com/lipis/gae-init/tree/lipis-rest/main/api

**Comment by gmist** [comment]
Looks good

**Comment by lipis** [comment]
Speaking of squashes.. I tried this in a test project but I want to test it with an existing one :) So could you squash those and most likely I will merge it the coming week!! Bunch of people are pushing me with this one and I think version 2 is coming in February :)

**Comment by lipis** [comment]
Thanks!

**Comment by topless** [comment]
+10

File: main/static/src/script/common/service.coffee [hunk 2]
```diff
@@ -1,4 +1,4 @@
-window.service_call = (method, url, params, data, callback) ->
+window.api_call = (method, url, params, data, callback) ->
```
Comment (lipis) [inline main/static/src/script/common/service.coffee]: Since we are breaking anyway... `service.coffee` ---> `api.coffee`

**Comment by lipis** [comment]
It looks like this is it :)

1. Squash
2. Update the `main/__init__.py` to version `2.0.0`
3. Make a new commit with the message `Version 2.0`
4. I'll press the magic green button :dancer: :dancers: :dancer:

**Comment by lipis** [comment]
:godmode:

**Pull request merged**